core/src/test/scala/unit/kafka/log/LogCleanerManagerTest.scala

```diff
@@ -177,6 +177,36 @@ class LogCleanerManagerTest extends Logging {
     assertEquals(10L, filthiestLog.firstDirtyOffset)
   }
 
+  @Test
+  def testDirtyOffsetResetIfLargerThanActiveLogSegmentBaseOffset(): Unit = {
```

Comment:
This is a good test case to have. This scenario could happen if there is an unclean leader election. Another useful test case is this one: https://github.com/hachikuji/kafka/commit/2c2e03743268e1e1b8896a31ae5c326c24d0326c#diff-c528f7f296be19c3144bd14a2d1e63aeR217. This checks the scenario I described above.